src/player/SurvivalBlockBreakHandler.php

```diff
@@ -65,8 +67,27 @@ private function calculateBreakProgressPerTick() : float{
 		if(!$this->block->getBreakInfo()->isBreakable()){
 			return 0.0;
 		}
-		//TODO: improve this to take stuff like swimming, ladders, enchanted tools into account, fix wrong tool break time calculations for bad tools (pmmp/PocketMine-MP#211)
-		$breakTimePerTick = $this->block->getBreakInfo()->getBreakTime($this->player->getInventory()->getItemInHand()) * 20;
+		$breakTimePerSecond = $this->block->getBreakInfo()->getBreakTime($this->player->getInventory()->getItemInHand());
+
+		if($this->player->isUnderwater()){ //TODO: Aqua Affinity enchantment
+			$breakTimePerSecond *= 5;
+		}
+		if(!$this->player->onGround){
+			$breakTimePerSecond *= 5;
+		}
+		if($this->player->canClimb() && $this->player->getWorld()->getBlock($this->player->getPosition())->canClimb()){
+			$breakTimePerSecond *= 5;
+		}
+
+		$effects = $this->player->getEffects();
+		if(($miningFatigue = $effects->get(VanillaEffects::MINING_FATIGUE())) !== null){
+			$breakTimePerSecond *= $breakTimePerSecond / ($breakTimePerSecond * 0.2 ** $miningFatigue->getEffectLevel());
```

Comment:
How confident are you in this? Last time I checked, the values were different for Bedrock, which is why I didn't implement this properly to begin with.

Reply:
```php
$breakTimePerTick = ceil($this->block->getBreakInfo()->getBreakTime($this->player->getInventory()->getItemInHand()) * 20);
       
        $breakTimePerTick *= 1 - (0.3 * $this->player->getEffects()->get(VanillaEffects::MINING_FATIGUE())?->getAmplifier() ?? 0);
```

Reply:
I don't know where is your code from but according to the wiki:
```
The mining speed is reduced to 0.30^level of the normal mining speed. For the in-game default level III, the speed is reduced to 0.33 (0.027×) of normal speed, taking 37 times longer than usual to break a block with the proper tool for that block. 
```
https://minecraft.fandom.com/wiki/Mining_Fatigue